synapse/handlers/register.py

```diff
@@ -149,7 +162,7 @@ def register(
                     user_id = None
                     token = None
                     attempts += 1
-                    if attempts > 5:
+                    if attempts > 50:
```

Comment:
But you're attempting to pick a starting point which is higher than any existing IDs? Or is it deliberate that you will re-issue any IDs which end up being a hole in the existing allocation? If the latter, 50 sounds like way too few.